### PR TITLE
[7.0][dotnet][Machine-learning][TCSACR-493]Add Tizen.MachineLearning.Train documentation

### PR DESCRIPTION
**Pull request by songgot**
[7.0][dotnet][Machine-learning][TCSACR-493]Add Tizen.MachineLearning.Train documentation
    
- Add overview.md newly
- Change overview.md to machine-learning-inference.md
- Add machine-learning-train.md
- Modify toc_all.md

Signed-off-by: hyunil park <hyunil46.park@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: docs/application/dotnet/guides/machine-learning/machine-learning-train.md [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+  model.Run("batch_size=9", "epochs=20");
+}
+catch (Exception e)
+{
+  // handle error
+}
+```
+## Destroy the model
+
+Model will be disposed by GC but you can use Dispose().
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/machine-learning/machine-learning-train.md:747]:
```suggestion
Model will be disposed by GC, but you can use Dispose().
```

File: docs/application/dotnet/guides/machine-learning/machine-learning-train.md [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+You can use the trained model with pipeline API as follows:
+
+```csharp
+using Tizen.MachineLearning.Inference;
+
+/* framework is 'nntrainer'*/
+const string description = "appsrc ! other/tensor,dimension=(string)2:1:1:1,type=(string)  int8,framerate=(fraction)0/1 ! tensor_filter framework=nntrainer model=model.ini ! tensor_sink";
+
+var pipeline = new Pipeline(description);
+```
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/machine-learning/machine-learning-train.md:778]: This md file (`machine-learning-train.md`) is missing a Related Information section mentioning dependencies.

**Review by safir-srbd — COMMENTED** [review]
@songgot Hello. Hope you are well.

I have gone through the files in this PR and have suggested some changes for you. Please incorporate them when possible.

File: docs/application/dotnet/guides/machine-learning/machine-learning-train.md [hunk 3]
```diff
@@ -0,0 +1,782 @@
...
+| &#xfeff;                       | beta2         | (float) | Beta2 coefficient for Adam                     |
+| &#xfeff;                       | epsilon       | (float) | Epsilon coefficient for Adam                   |
+
+### Dataset class
+
+`Tizen.MachineLearning.Train.Dataset` class is in charge of feeding data into the model.
+The dataset can be created from a file.
+For more information, see [configure the model](#configure-the-model) section.
+
+The Following code is an example of handling `Tizen.MachineLearning.Train.Dataset` class:
```
Comment (safir-srbd) [inline docs/application/dotnet/guides/machine-learning/machine-learning-train.md:416]:
```suggestion
The following code is an example of handling `Tizen.MachineLearning.Train.Dataset` class:
```

**Review by safir-srbd — COMMENTED** [review]
@songgot Thank you for incorporating the suggested changes. After another review, I had one small suggestion. Please incorporate that.

File: docs/application/dotnet/guides/machine-learning/machine-learning-train.md (hunk 3 above)
Comment (songgot) [reply docs/application/dotnet/guides/machine-learning/machine-learning-train.md:416]: thank you.

**Review by safir-srbd — APPROVED** [review]
@songgot Thank you for incorporating the suggested change.

@asm-foysal @mijin85cho I am approving this PR for merging.

**Review by mijin85cho — COMMENTED** [review]
Hello, I just briefly skimmed the update  and got below questions.
Would you please check the comments?

File: docs/application/toc_all.md [hunk 4]
```diff
@@ -127,8 +127,10 @@
 
 ### Machine Learning
 #### [Overview](/application/dotnet/guides/machine-learning/overview.md)
+#### [Machine Learning Inference](/application/dotnet/guides/machine-learning/machine-learning-inference.md)
 #### [Single Shot](/application/dotnet/guides/machine-learning/singleshot.md)
 #### [Pipeline](/application/dotnet/guides/machine-learning/pipeline.md)
```
[the remark this answers is not in the document]
Comment (songgot) [reply docs/application/toc_all.md:132]: okay

File: docs/application/dotnet/guides/machine-learning/machine-learning-train.md [hunk 5]
```diff
@@ -0,0 +1,778 @@
...
+
+> [!NOTE]
+> Every example code does not handle all error use cases.
+> The error must be handled more extensively compared to the example code written on this page.
+
+## Prerequisites
+
+To enable your application to use `Tizen.MachineLearning.Train` class:
+
+1. To use the method and properties of the 'Tizen.MachineLearning.Train.Model', 'Tizen.MachineLearning.Train.Dataset', 'Tizen.MachineLearning.Train.Layer' and 'Tizen.MachineLearning.Train.Optimizer', include the 'Tizen.MachineLearning.Train' namespace in your application:
```
[the remark this answers is not in the document]
Comment (songgot) [reply docs/application/dotnet/guides/machine-learning/machine-learning-train.md:22]: thank you

File: docs/application/toc_all.md [hunk 6]
```diff
@@ -127,8 +127,10 @@
 
 ### Machine Learning
 #### [Overview](/application/dotnet/guides/machine-learning/overview.md)
-#### [Single Shot](/application/dotnet/guides/machine-learning/singleshot.md)
-#### [Pipeline](/application/dotnet/guides/machine-learning/pipeline.md)
+#### [Machine Learning Inference](/application/dotnet/guides/machine-learning/machine-learning-inference.md)
+##### [Single Shot](/application/dotnet/guides/machine-learning/singleshot.md)
+##### [Pipeline](/application/dotnet/guides/machine-learning/pipeline.md)
```
[the remark this answers is not in the document]
Comment (songgot) [reply docs/application/toc_all.md]: thank you.

**Comment by mijin85cho** [comment]
@safir-srbd @asm-foysal My comments are also resolved. From next time please don't forget to check the working of LNB, too

**Pull request merged**